modules/ocf/manifests/ssl/lets_encrypt.pp

```diff
@@ -0,0 +1,93 @@
+# TODO: Separate out this file into the http and dns versions, since they can't
+# overlap anyways due to resource duplicate declaration problems
+define ocf::ssl::lets_encrypt(
+  Array[String] $domains = [$::fqdn],
+  Enum['http', 'dns'] $challenge_type = 'dns',
+) {
+  if $challenge_type == 'http' {
+    package { ['acme-tiny', 'python3-openssl']:; }
+
+    file {
+      [
+        '/var/lib/lets-encrypt/.well-known',
+        '/var/lib/lets-encrypt/.well-known/acme-challenge',
+      ]:
+        ensure  => directory,
+        owner   => ocfletsencrypt,
+        group   => ssl-cert,
+        require => Package['ssl-cert'];
+
+      '/usr/local/bin/ocf-lets-encrypt':
+        source  => 'puppet:///modules/ocf/ssl/ocf-lets-encrypt',
+        mode    => '0755',
+        require => Package['acme-tiny', 'python3-openssl'];
+    }
+  } else {
+    ocf::repackage { 'dehydrated':
+      backport_on => ['jessie', 'stretch'],
+    }
+    package { 'dehydrated-hook-ddns-tsig':
+      require => Ocf::Repackage['dehydrated'],
+    }
+
+    $letsencrypt_ddns_key = assert_type(Stdlib::Base64, hiera('letsencrypt::ddns::key'))
+
+    # TODO: Move these somewhere else so this defined resource can be used
```

Comment:
No, like the certs directory can't really be moved unless we move the whole base directory, I meant to move this inside puppet to a different manifest so it could be included or something instead of redefined every time. This stuff only needs to be declared once, although I'm not sure how the domains list will work, it'll probably need to use concat or something.